Zotero/Scenes/Detail/PDF/Views/PDFAnnotationsViewController.swift

```diff
@@ -66,13 +68,18 @@ final class PDFAnnotationsViewController: UIViewController {
     override func viewIsAppearing(_ animated: Bool) {
         super.viewIsAppearing(animated)
 
-        var snapshot = NSDiffableDataSourceSnapshot<Int, PDFReaderState.AnnotationKey>()
-        snapshot.appendSections([0])
-        snapshot.appendItems(viewModel.state.sortedKeys)
         tableView.setEditing(viewModel.state.sidebarEditingEnabled, animated: false)
-        dataSource.apply(snapshot, animatingDifferences: false)
-        if let key = viewModel.state.focusSidebarKey, let indexPath = dataSource.indexPath(for: key) {
-            tableView.selectRow(at: indexPath, animated: false, scrollPosition: .middle)
+        updateQueue.async { [weak self] in
+            guard let self else { return }
+            var snapshot = NSDiffableDataSourceSnapshot<Int, PDFReaderState.AnnotationKey>()
+            snapshot.appendSections([0])
+            snapshot.appendItems(viewModel.state.sortedKeys)
+            dataSource.apply(snapshot, animatingDifferences: false)
+            if let key = viewModel.state.focusSidebarKey, let indexPath = dataSource.indexPath(for: key) {
+                inMainThread { [weak self] in
+                    self?.tableView.selectRow(at: indexPath, animated: false, scrollPosition: .middle)
+                }
+            }
```

Comment:
```suggestion
            dataSource.apply(snapshot, animatingDifferences: false)
            dataSource.apply(snapshot, animatingDifferences: false) { [weak self] in
                guard let self else { return }
                if let key = viewModel.state.focusSidebarKey, let indexPath = dataSource.indexPath(for: key) {
                    tableView.selectRow(at: indexPath, animated: false, scrollPosition: .middle)
                }
            }
```

Reply:
Removed unnecessary main thread in 5e82db4a5bb6dde2b64def6d517e70c00e6a93b3